tuf/ngclient/updater.py

```diff
@@ -471,7 +503,67 @@ def _preorder_depth_first_walk(
         # If this point is reached then target is not found, return None
         return None
 
+    def _download_directory(self, folder: str) -> None:
+        # Downloading all metadata inside the specified folder
+
+        path = (
+            f"{BASE_URL}/repository_data/TAP 14"  # File path to download from
+        )
+        path = os.path.join(path, folder)  # Add the folder to the path
+
+        # Repository might be storing data in different formats.
+
+        files = next(os.walk(path))[2]  # Get all the files inside the list
+        for file in files:
+            if file.endswith(".json"):
+                self._download_metadata(file)
+
 
 def _ensure_trailing_slash(url: str) -> str:
     """Return url guaranteed to end in a slash"""
     return url if url.endswith("/") else f"{url}/"
+
+
+def _get_spec_version(
+    repository_versions: List[str],
+    spec_version: str,
+    supported_versions: List[str],
+) -> str:
+    """Returns the specification version to be used."""
```

Comment:
Make sure to mention the return type, which is a little funky. When does it return the message?

Reply:
How about? 
```
Returns the specification version to be used and displays a warning if chosen spec_version
is lower than the highest repository version.
```

Reply:
Looks great! I might add "specification version to be used, following the rules of TAP-14"

Reply:
Also, shouldn't I mention the return type in the function definition above and keep the docstring to just an explanation?

Reply:
I'm a bit confused. If `SpecificationVersion` would essentially be derived from `str` I'm still comparing it with `latest_repo_version` and performing int type functions on it. Won't it throw a `TypeError` in that case?

Reply:
Right—if you just make `SpecificationVersion` as a `NewType`, you can't compare them directly. It's the same as what you have now.

*Eventually*, you'd make a full `class SpecificationVersion` which doesn't require conversion to/from int.